presto-hive/src/main/java/com/facebook/presto/hive/HiveWriterFactory.java

```diff
@@ -494,6 +494,9 @@ private WriterParameters getWriterParametersForExistingPartitionedTable(String p
     {
         switch (insertExistingPartitionsBehavior) {
             case APPEND:
+                if (MetastoreUtil.isPrestoMaterializedView(table)) {
```

Comment:
I think we should keep materialized view insert partition logic independent of insertExistingPartitionsBehavior. With this, it violates few condition checks defined at the initialization time.

We should check for MV outside the switch for insertExistingPartitionsBehavior.